### PR TITLE
Second problem

### PR DESCRIPTION
**Pull request by HiciuStefan**

### CODE REVIEW

File: second_problem.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class SecondProblem
+
+  attr_reader :min_value
+
+  def initialize(numbers_array)
+    @numbers_array = numbers_array
+    Struct.new("MinMax", :second_large, :first_large)
+    @min_value = -1
```
Comment (alexandru-calinoiu) [inline second_problem.rb]: I don't think this is correct, integers have negative numbers smaller than -1, so for example an array of [-2, -3, -4] should return [-2, -3].

In the initial solution you have a large negative number, I wanted to know how you arrived at that number.

**Review by alexandru-calinoiu — CHANGES_REQUESTED** [review]
Please review my comment.

File: second_problem.rb (hunk 1 above)
Comment (HiciuStefan) [reply second_problem.rb]: My ideea was to get a "big" number to have a kind of -MAXINT. That's why the -1234567890.
In this iteration i just considered positive numbers. -> bad
I've updated the code to have this -MAXINT value.

File: second_problem.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+class SecondProblem
+
+  N_BYTES = [42].pack('i').size
+  N_BITS = N_BYTES * 16
+  MAX = 2 ** (N_BITS - 2) - 1
```
Comment (alexandru-calinoiu) [inline second_problem.rb:5]: This makes sens :+1:

**Comment by alexandru-calinoiu** [comment]
Thank you, will write a email shortly.

**Pull request closed without merging**